dynamic_analysis_pkg/dynamic_analysis_pkg/__init__.py

```diff
@@ -0,0 +1,157 @@
+    JOB_BASE = os.path.join(UNICORE_ENDPOINT, 'jobs')
+
+    def __init__(self, token, sim_id, base_working_directory=None, new_circuit_path=None):
+
+        if base_working_directory is None:
+            base_working_directory = self.DEFAULT_WD_BASE
+        if new_circuit_path is None:
+            new_circuit_path = self.DEFAULT_CIRCUIT_DIR
+
+        self.files_list = None
+
+        self.circuit_dir = Path(new_circuit_path)
+        if not self.circuit_dir.is_dir():
+            raise ValueError('[ERROR] Circuit path does not exists')
+
+        self.local_dir = Path(base_working_directory) / sim_id
+        self.local_dir.mkdir(parents=True, exist_ok=True)
+
+        self.hpc_dir_path = self.HPC_DIR_BASE / sim_id
+
+        self.fetch_results(token, sim_id)
+        self.blueconfig = self.local_dir / 'BlueConfig'
+
+    def fetch_results(self, token, sim_id):
+        self.retrieve_sim_info(token, sim_id)
+        self.get_sim_results()
+
+    def retrieve_sim_info(self, token, sim_id=None):
+        tr = unicore_client.Transport(token)
+        job_url = os.path.join(self.JOB_BASE, sim_id)
+        job = unicore_client.Job(tr, job_url)
+        job_name = job.properties['name']
+        L.info('Fetching results of [%s]. Please wait ...', job_name)
+        storage = job.working_dir
+        self.files_list = storage.listdir()
+
+    def get_sim_results(self):
+        self.download_file_to_storage('BlueConfig')
+        self.download_report()
+        self.download_file_to_storage('out.dat')
+        L.info('Result were saved at: %s', self.local_dir)
+
+    def download_file_to_storage(self, file_name):
+        ''' function to download the file and add it to kernel file system '''
+        file_output_path = self.local_dir / file_name
+        if file_output_path.exists():
+            L.info('- [%s] file already exists. Skipping download.', file_name)
+            return
+
+        L.info('- Fetching [%s] ...', file_name)
+        try:
+            current_file = self.files_list[file_name]
+        except:
+            raise KeyError('The file [{}] is not present on the simulation results'.format(file_name))
+
+        if file_name == 'BlueConfig':
+            file_content = current_file.raw().read()
+            bc_str = file_content.decode('utf-8')
+            writable_content = bc_str.replace(self.ORIGIN_CIRCUIT_PATH, self.circuit_dir)
+            writable_content = writable_content.replace(str(self.hpc_dir_path), str(self.local_dir))
+
+            with open(file_output_path, 'w') as fd:
+                fd.write(writable_content)
+
+        else:
+            current_file.download(file_name) # moves to home always
+            move(file_name, file_output_path)
+
+        L.info('- [%s] downloaded', file_name)
+
+    def download_report(self):
+        reports = [x for x in self.files_list if x.endswith('.bbp')]
+        if not reports:
+            L.info('No reports were found')
+            return
```

Comment:
You don't need the `return`
The for loop will be skipped as `reports` is empty.